policy/examples/tiered-parking-fees-per-hour.json

```diff
@@ -0,0 +1,57 @@
+{
+  "updated": 0,
+  "version": "1.2.0",
+  "data": {
+    "policy": [
+      {
+        "name": "Tiered Dwell Time Example",
+        "description": "First hour $2, second hour $4, every hour onwards $10",
+        "policy_id": "2800cd0a-7827-4110-9713-b9e5bf29e9a1",
+        "start_date": 1558389669540,
+        "publish_date": 1558389669540,
+        "end_date": null,
+        "prev_policies": null,
+        "provider_ids": [],
+        "currency": "USD",
+        "rules": [
```

Comment:
I think that the methodology you described functionally works how I'd expect at the end of the day, and the logic checks out! However, I have perhaps a slightly more intuitive way of thinking about it. The most intuitive way I can think of evaluating policies with rates that specify `each_time_unit`, is by treating `each_time_unit` tick as an event (note, not an MDS event, more of an event-driven-system kind of event). This would enable continuous charging throughout the duration of the parking, as opposed to waiting till the absolute duration of the park has been completed.

Let's, for the sake of making this a bit easier to reason about, assume that the `time_unit` for each rule is the same, so we can evaluate the whole policy at once every hour on the dot. Because this is `each_time_unit` instead of `each_complete_time_unit`, we will evaluate on first entry in addition to after each hour ticks.

# Continuous(ish) Evaluation Methodology for this Policy
I'm gonna approach this with some brevity here, but if you'd like me to dial into some more detail please let me know and I'm happy to!
## @00:00 (right when first parked)
* First rule doesn't match, continue 
* Second rule doesn't match, continue
* Third rule matches, charge **$2**

## @01:00
* First rule doesn't match, continue
* Second rule matches, charge **$4**, short-circuit execution

## @02:00
* First rule matches, charge **$10**, short-circuit execution

## @03:00
* First rule matches, charge **$10**, short-circuit execution

## @04:00
* First rule matches, charge **$10**, short-circuit execution

## @05:00
* First rule matches, charge **$10**, short-circuit execution

## @06:00
* First rule matches, charge **$10**, short-circuit execution


At the end of the day, there will be a charge for $2 + $4 + $10 + $10 + $10 + $10 + $10 = $56.

Reply:
In order for this to work the the comparison to `maximum` looks like it needs to be `>=`? And same for `minimum`?

If you did have different units in the rules would you pre-process the rules into the same units in order to figure out your clock-ticks?

Reply:
Yeah, `>=` for maximums, and similarly `<=` for minimums I think (let me update my other comment, I think I didn't include the `=` component).

Reply:
If I had different units on the rules I'd probably specify an interval to re-evaluate of the lowest-duration unit referenced (e.g. if there were some rules with minutes and some with hours, I'd evaluate on a minutely basis). I'd then probably add some logic to ensure that double-fees won't get charged if a high-duration rule is hit again within that unit of its last occurrence, and instead results in a short-circuit without a charge.

```
if (current_timestamp - prev_fee_timestamp < matched_rule.unit) {
  // don't charge anything, because we don't want to double charge
}
```